### PR TITLE
openstack: ovh renamed flavors : s/eg/hg/

### PR DESCRIPTION
**Pull request by dachary**
http://travaux.ovh.net/?do=details&id=17532

Signed-off-by: Loic Dachary <loic@dachary.org>

### CODE REVIEW

**Comment by smithfarm** [comment]
```
smithfarm@f6f2e737b671:~$ openstack flavor list
+--------------------------------------+----------------+--------+------+-----------+-------+-----------+
| ID                                   | Name           |    RAM | Disk | Ephemeral | VCPUs | Is Public |
+--------------------------------------+----------------+--------+------+-----------+-------+-----------+
| 068f098e-e1b1-4f31-8441-ad3c0eeacc72 | hg-7-ssd       |   7000 |  100 |         0 |     2 | True      |
| 0c36bb1d-bd1e-4294-ac75-60fd727054c8 | win-hg-15      |  15000 |  400 |         0 |     4 | True      |
| 0e2b8841-d764-49b4-b2ac-24c8df2c6806 | sp-30          |  30000 |  200 |         0 |     2 | True      |
| 0fd97bb7-8834-4d59-bb8a-3d0ab28a65cd | win-sp-60-ssd  |  60000 |  200 |         0 |     4 | True      |
| 169e7020-0436-4e8c-b2c8-213a57a044b5 | win-sp-120     | 120000 |  800 |         0 |     8 | True      |
| 18811d09-053e-4ad0-b297-3c11ceefd136 | sp-60          |  60000 |  400 |         0 |     4 | True      |
| 1e2c13fc-4ba9-4773-a13e-75ce96fb4e96 | hg-120         | 120000 | 1600 |         0 |    32 | True      |
| 1e5a7cf5-d58c-4633-8069-5aba49b00be1 | win-sp-240     | 240000 | 1600 |         0 |    16 | True      |
| 22096e9d-5150-4f54-bb8d-6c8b243a28bb | win-sp-60      |  60000 |  400 |         0 |     4 | True      |
| 229eabf4-ada1-49ee-8bde-f1c574786fd3 | hg-60-ssd      |  60000 |  600 |         0 |    16 | True      |
| 23ec7924-483c-4414-9aaf-bf03575f0bfe | hg-15          |  15000 |  400 |         0 |     4 | True      |
| 24c1f706-3278-4ca7-a773-7df7f999cf14 | win-hg-60      |  60000 | 1600 |         0 |    16 | True      |
| 2f6e1384-f8c0-463d-bc10-3c9178d30385 | hg-120-ssd     | 120000 |  800 |         0 |    32 | True      |
| 30b43ca1-131f-4b0d-8087-e03bc99f7ec6 | sp-120         | 120000 |  800 |         0 |     8 | True      |
| 3502c448-8c96-4062-bf78-c5d2ac962e6d | win-hg-120     | 120000 | 1600 |         0 |    32 | True      |
| 3a65f02e-69a5-4055-8688-f553b5d5a4fd | win-sp-30-ssd  |  30000 |  100 |         0 |     2 | True      |
| 3a9711d2-a4fb-4d2f-a3cd-1972fd53eb5e | vps-ssd-3      |   8000 |   40 |         0 |     2 | True      |
| 3fe893ce-adce-4113-8639-88a583b4859f | win-hg-60-ssd  |  60000 |  600 |         0 |    16 | True      |
| 5bb302e6-e9be-46af-a39b-6ad5ce58cec0 | win-hg-120-ssd | 120000 |  800 |         0 |    32 | True      |
| 66093206-63a6-40dd-b9e7-6398e849dce7 | hg-30-ssd      |  30000 |  400 |         0 |     8 | True      |
| 6763bd1b-5d23-4c87-aca7-d46e35d343a4 | win-hg-7       |   7000 |  200 |         0 |     2 | True      |
| 74204dc3-024f-410d-8a0b-3210c48f4bbe | sp-120-ssd     | 120000 |  400 |         0 |     8 | True      |
| 7eacbe7c-5033-4123-8eec-d61bf27359f9 | win-sp-30      |  30000 |  200 |         0 |     2 | True      |
| 8485c632-47c4-429a-8305-01b0fbce2fce | win-sp-120-ssd | 120000 |  400 |         0 |     8 | True      |
| 8f08b006-a68b-4eed-80cf-36f8212f8d73 | win-hg-30      |  30000 |  800 |         0 |     8 | True      |
| 9153a1bd-0053-4d9d-a564-28e1317d3a5e | win-hg-7-ssd   |   7000 |  100 |         0 |     2 | True      |
| 98c1e679-5f2c-4069-b4da-4a4f7179b758 | vps-ssd-1      |   2000 |   10 |         0 |     1 | True      |
| 9c2011f2-6846-49af-8dc5-813e2a37584c | win-sp-240-ssd | 240000 |  800 |         0 |    16 | True      |
| a42a6051-16c7-4563-9cca-9c1abf36e460 | sp-240         | 240000 | 1600 |         0 |    16 | True      |
| a7bea170-bdc4-4d0a-89b6-76736d63ec81 | hg-15-ssd      |  15000 |  200 |         0 |     4 | True      |
| aa9e1a75-fcb4-421d-a9c2-7c30bbdbbb12 | sp-240-ssd     | 240000 |  800 |         0 |    16 | True      |
| b41c89be-7b7d-4ecc-9cd8-bea145bbbd76 | hg-7           |   7000 |  200 |         0 |     2 | True      |
| b62d1735-774d-4983-9ed6-283c921d29d7 | hg-30          |  30000 |  800 |         0 |     8 | True      |
| bc2a9079-8199-4e96-aec5-7d6c93ec79f3 | sp-60-ssd      |  60000 |  200 |         0 |     4 | True      |
| c3d1d22a-03b3-4fa4-bcc4-1dff4e2b2d32 | vps-ssd-2      |   4000 |   20 |         0 |     1 | True      |
| ccb922b6-04ad-422d-a8bc-7aff1eae6954 | hg-60          |  60000 | 1600 |         0 |    16 | True      |
| e48b2175-06d4-46e0-a99c-a877f1233434 | win-hg-15-ssd  |  15000 |  200 |         0 |     4 | True      |
| ea9aea13-5660-4811-bf70-aa5082ad7d1f | sp-30-ssd      |  30000 |  100 |         0 |     2 | True      |
| f893c6f6-4394-430f-8cad-270dd884dd8e | win-hg-30-ssd  |  30000 |  400 |         0 |     8 | True      |
+--------------------------------------+----------------+--------+------+-----------+-------+-----------+
```

**Comment by smithfarm** [comment]
@dachary It looks like the following OVH flavors match the new regex:

```
smithfarm@wilbur:~/suse/storage/ovh> grep -E ' (vps|hg)-.*-ssd' flavors.txt
| 068f098e-e1b1-4f31-8441-ad3c0eeacc72 | hg-7-ssd       |   7000 |  100 |         0 |     2 | True      |
| 229eabf4-ada1-49ee-8bde-f1c574786fd3 | hg-60-ssd      |  60000 |  600 |         0 |    16 | True      |
| 2f6e1384-f8c0-463d-bc10-3c9178d30385 | hg-120-ssd     | 120000 |  800 |         0 |    32 | True      |
| 66093206-63a6-40dd-b9e7-6398e849dce7 | hg-30-ssd      |  30000 |  400 |         0 |     8 | True      |
| a7bea170-bdc4-4d0a-89b6-76736d63ec81 | hg-15-ssd      |  15000 |  200 |         0 |     4 | True      |
```

I guess that's sufficient to satisfy our requirements in `buildpackages.py`?

**Comment by dachary** [comment]
Hum, it does not match vps-ssd-1, really ?

**Comment by smithfarm** [comment]
Nope. But this one does:

```
smithfarm@wilbur:~/suse/storage/ovh> grep -E ' (hg-[0-9]+-ssd|vps-ssd-[0-9]+)' flavors.txt
| 068f098e-e1b1-4f31-8441-ad3c0eeacc72 | hg-7-ssd       |   7000 |  100 |         0 |     2 | True      |
| 229eabf4-ada1-49ee-8bde-f1c574786fd3 | hg-60-ssd      |  60000 |  600 |         0 |    16 | True      |
| 2f6e1384-f8c0-463d-bc10-3c9178d30385 | hg-120-ssd     | 120000 |  800 |         0 |    32 | True      |
| 3a9711d2-a4fb-4d2f-a3cd-1972fd53eb5e | vps-ssd-3      |   8000 |   40 |         0 |     2 | True      |
| 66093206-63a6-40dd-b9e7-6398e849dce7 | hg-30-ssd      |  30000 |  400 |         0 |     8 | True      |
| 98c1e679-5f2c-4069-b4da-4a4f7179b758 | vps-ssd-1      |   2000 |   10 |         0 |     1 | True      |
| a7bea170-bdc4-4d0a-89b6-76736d63ec81 | hg-15-ssd      |  15000 |  200 |         0 |     4 | True      |
| c3d1d22a-03b3-4fa4-bcc4-1dff4e2b2d32 | vps-ssd-2      |   4000 |   20 |         0 |     1 | True      |
```

File: tasks/buildpackages.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def task(ctx, config):
         openstack = OpenStack()
         openstack.set_provider()
         if openstack.provider == 'ovh':
-            select = '^(vps|eg)-'
+            select = '^(vps|hg)-.*-ssd'
```
Comment (smithfarm) [inline tasks/buildpackages.py]: `select = '^(hg-[0-9]+-ssd|vps-ssd-[0-9]+)'`

**Comment by dachary** [comment]
@smithfarm good catch. There was an extra - in the regexp. Amended and repushed.
<pre>
>>> import re
>>> re.match('^(vps|hg)-.*ssd', 'hg-30-ssd')
<_sre.SRE_Match object at 0x7fb0129e2300>
>>> re.match('^(vps|hg)-.*ssd', 'vps-ssd-2')
<_sre.SRE_Match object at 0x7fb0129e2198>
>>> 
</pre>

**Comment by smithfarm** [comment]
@dachary I guess this needs to be cherry-picked to jewel, huh?

**Pull request merged**